### PR TITLE
Add in standard tests to rqt_shell.

### PR DESCRIPTION
**Pull request by clalancette**
That way we know it conforms to our standards.

### CODE REVIEW

**Comment by clalancette** [comment]
Pulls: https://github.com/ros-visualization/rqt_shell/pull/24
Gist: https://gist.githubusercontent.com/clalancette/c38e8bdb294dadf6aff1f9314fb845cb/raw/eb198c3dae8dbd9dd0ddeac787adc624aca94b65/ros2.repos
BUILD args:  --packages-above-and-dependencies rqt_shell
TEST args:  --packages-above rqt_shell
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/15008
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22323/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16586/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1854/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/23114/)

File: src/rqt_shell/spyder_shell_widget.py [hunk 1]
```diff
@@ -30,14 +28,14 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+import os
+
+from python_qt_binding.QtCore import QProcess, QTextCodec, SIGNAL, Signal
```
Comment (gunasekaran-goatrobotics) [inline src/rqt_shell/spyder_shell_widget.py:33]: Is there any purpose for changing the order when importing?

File: src/rqt_shell/xterm_widget.py [hunk 2]
```diff
@@ -45,7 +41,7 @@ class XTermWidget(QX11EmbedContainer):
     def __init__(self, parent=None, script_path=None):
         # possible use os to do better justice to script path??
         if script_path:
-            xterm_cmd += " -e $SHELL -c 'source %s; $SHELL'" % os.path.abspath(script_path)
+            self.xterm_cmd += " -e $SHELL -c 'source %s; $SHELL'" % os.path.abspath(script_path)
```
Comment (gunasekaran-goatrobotics) [inline src/rqt_shell/xterm_widget.py:44]: can we change the `self. xterm_cmd` variable name to share the difference between the local variable and the instance variable?

**Review by gunasekaran-goatrobotics — COMMENTED** [review]
code style improvements

File: src/rqt_shell/spyder_shell_widget.py (hunk 1 above)
Comment (clalancette) [reply src/rqt_shell/spyder_shell_widget.py:33]: Yes, it is to satisfy the linters, which expect that they are in alphabetical order.

File: src/rqt_shell/xterm_widget.py (hunk 2 above)
Comment (clalancette) [reply src/rqt_shell/xterm_widget.py:44]: Hm, sorry, I'm not sure what you are asking here.  Can you rephrase your question?

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-news-for-the-week-of-december-30th-2024-2025/41389/1

File: src/rqt_shell/xterm_widget.py (hunk 2 above)
Comment (gunasekaran-goatrobotics) [reply src/rqt_shell/xterm_widget.py:44]: we have `xterm_cmd` as a class variable and `self.xterm_cmd` as an instance variable with the same name. So we can change the name of either variable, which is good for understanding.

Comment (xguay) [reply src/rqt_shell/xterm_widget.py:44]: I know I'm coming out of nowhere. I'm not even sure how I got to this PR, but figured I'd review it anyway.

That original line of code shouldn't work. I ran a little test and this should be throwing `UnboundLocalError: local variable 'xterm_cmd' referenced before assignment` without the `self.`.

I'd recommend putting this change in it's own PR because it's a bug fix and outside the scope of the documentation changes the rest of this is about.

**Review by xguay — COMMENTED** [review]
Nothing I'd stop you from merging, but I do still recommend putting the `self.` change in it's own PR so it can be cherry-picked or copied to other places if necessary.

**Review by xguay — APPROVED** [review]

**Review by ahcorde — APPROVED** [review]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22357/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16620/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1888/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/23148/)
